pkg/generate/app/cmd/resolve.go

```diff
@@ -116,7 +119,7 @@ func Resolve(r *Resolvers, c *ComponentInputs, g *GenerationInputs) (*ResolvedCo
 	}
 
 	if g.Strategy != generate.StrategyUnspecified && len(repositories) == 0 && !g.BinaryBuild {
-		return nil, errors.New("when --strategy is specified you must provide at least one source code location")
+		return nil, errors.New("when --strategy is specified you must provide at least one source code location; see the argument classification results above for possible explanations a source code location was not found")
```

Comment:
explanations -> explanations why

I'd be inclined to be more parsimonious than friendly here, and drop this change.  For example, I don't think it's assured at this point that there will be explanations above (e.g. run `oc new-app --strategy=docker`).